### PR TITLE
feat: edit blog post

### PR DESCRIPTION
**Pull request by AMdabour**
## Description

Implemented and tested the feature for updating blog posts at the endpoint `api/v1/blog/{id}`. This feature allows authenticated users to update the title and content of their blog posts. Includes improvements in handling authentication and error responses.

### Changes Made:
- Added functionality to the `api/v1/blog/{id}` endpoint to allow updating of blog posts.
- Integrated token-based authentication to secure the endpoint.
- Implemented validation to handle empty fields and non-existent blog posts.
- Added detailed error responses for various scenarios including invalid tokens and non-existent posts.
- Updated tests to ensure proper functionality and error handling.

## Motivation and Context

The update feature for blog posts enhances the application's content management capabilities by allowing users to modify their posts. The authentication and validation improvements ensure that only authorized users can make changes and that input errors are handled gracefully.

## How Has This Been Tested?

- **Testing Environment:** Local development environment with `pytest` for unit and integration testing.
- **Tests Run:**
  - Verified successful updates to blog posts with valid tokens.
  - Ensured proper handling of non-existent blog posts with appropriate error responses.
  - Tested invalid token scenarios to confirm unauthorized access is denied.
  - Checked handling of empty fields to ensure validation errors are properly returned.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: main.py
Comment (joboy-dev) [inline main.py:1]: Why are you adding routes to the main.py file. It should be in api/v1/routes/__init__.py.

File: api/v1/services/blog_services.py
Comment (joboy-dev) [inline api/v1/services/blog_services.py:1]: The way this is set up is not right. Check the other service files to set this up properly

**Review by joboy-dev — CHANGES_REQUESTED** [review]
Make changes. Also why is blog id a uuid when all ids have been set to string with a default uuid attached.  So check your blog model

**Comment by AMdabour** [comment]
@joboy-dev 
i have made all changes you requested please review again

**Comment by CodewithSegNet** [comment]
Please wait for a mentor's review.

File: api/v1/routes/blog.py [hunk 1]
```diff
@@ -15,3 +19,26 @@ def get_all_blogs(db: Session = Depends(get_db)):
         return []
     return blogs
 
+blog = APIRouter(tags=["blogs"])
+
+@blog_router.put("/blog/{id}", response_model=blogUpdateResponseModel)
```
Comment (zxenonx) [inline api/v1/routes/blog.py:24]:
```suggestion
@blog_router.put("/blogs/{id}", response_model=blogUpdateResponseModel)
```

File: api/v1/schemas/blog.py [hunk 2]
```diff
@@ -3,6 +3,15 @@
 from uuid import UUID
 from datetime import datetime
 
+class blogRequest(BaseModel):
```
Comment (zxenonx) [inline api/v1/schemas/blog.py:6]:
```suggestion
class BlogRequest(BaseModel):
```

File: api/v1/schemas/blog.py [hunk 3]
```diff
@@ -3,6 +3,15 @@
...
+    title: str
+    content: str
+
+class blogUpdateResponseModel(BaseModel):
```
Comment (zxenonx) [inline api/v1/schemas/blog.py:10]:
```suggestion
class BlogUpdateResponseModel(BaseModel):
```

File: api/v1/routes/blog.py [hunk 4]
```diff
@@ -15,3 +19,26 @@ def get_all_blogs(db: Session = Depends(get_db)):
...
+async def update_blog(id: str, blogPost: blogRequest, db: Session = Depends(get_db), current_user: User = Depends(get_current_user)):
```
Comment (zxenonx) [inline api/v1/routes/blog.py:25]:
```suggestion
async def update_blog(id: str, blog_post: BlogRequest, db: Session = Depends(get_db), current_user: User = Depends(get_current_user)):
```

**Review by zxenonx — CHANGES_REQUESTED** [review]
Thanks @AMdabour 
I left some comments

File: api/v1/schemas/blog.py (hunk 3 above)
Comment (AMdabour) [reply api/v1/schemas/blog.py]: changed

**Comment by CodewithSegNet** [comment]
add a link to the issue you worked on in your PR

**Comment by AMdabour** [comment]
@zxenonx 
i have made all changes sir please review my PR again

**Review by zxenonx — APPROVED** [review]
Thanks @AMdabour

**Pull request merged**